local.c

```diff
@@ -1480,6 +1480,19 @@ static unsigned int is_global_attr(struct iio_channel *chn, const char *attr)
 
 	len = ptr - attr;
 
+	// Check if matching global differential attr,
+	// like "voltage-voltage"
+	dashptr = strchr(attr, '-');
+	if (dashptr && dashptr > attr && dashptr < ptr) {
+		unsigned int len1 = dashptr - attr,
+					 len2 = ptr - dashptr - 1;
+		if (chn->id[len1] >= '0' && chn->id[len1] <= '9' &&
```

Comment:
You should check that chn->id has at least len1 characters, otherwise you're reading garbage here.

Reply:
Yes! I realized that I left a buffer overrun. I was about to push another commit, but I had mistakenly assumed a single digit for X, and @rgetz reminded me that the channel index can go multiple digits. So I think I need to do two things:

1. Check that chn->id also has a dash, (if not, it's not a match at all)
2. Check that it's total length is the dash offset plus len2.

I should get an update pushed in a few hours.

Reply:
I'm not sure about checking for multiple digits. if a channel has a global attribute, it applies to voltage1 the same as voltage11 the same as voltage111.

as long as there is a single char that is a number at the end of "voltage", it should be fine. It doesn't matter how many char/numbers there are.

I think this is how the code works today...

Reply:
Understood. I just made an assumption about the position of the dash in the chn->id, by assuming there was one digit. This is obviously wrong, so I will fix it to search for the dash in the id and use that as the relative position to start the search for the second part.